plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/AIAssistantRegistry.java

```diff
@@ -16,17 +16,44 @@
  */
 package org.jkiss.dbeaver.model.ai;
 
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.IExtensionRegistry;
+import org.eclipse.core.runtime.Platform;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.utils.CommonUtils;
+
 public class AIAssistantRegistry {
-    private static final AIAssistantRegistry INSTANCE = new AIAssistantRegistry();
 
-    public static AIAssistantRegistry getInstance() {
-        return INSTANCE;
+    private static AIAssistantRegistry instance = null;
+
+    public synchronized static AIAssistantRegistry getInstance() {
+        if (instance == null) {
+            instance = new AIAssistantRegistry(Platform.getExtensionRegistry());
+        }
+        return instance;
     }
 
-    private AIAssistantRegistry() {
+    private final AIAssistantDescriptor customAssistant;
+
+    public AIAssistantRegistry(IExtensionRegistry registry) {
+        AIAssistantDescriptor customAssistantDescriptor = null;
+        IConfigurationElement[] extElements = registry.getConfigurationElementsFor("com.dbeaver.ai.assistant");
+        for (IConfigurationElement ext : extElements) {
+            if ("assistant".equals(ext.getName())) {
+                AIAssistantDescriptor descriptor = new AIAssistantDescriptor(ext);
+                if (!CommonUtils.isEmpty(descriptor.getReplaces())) {
+                    customAssistantDescriptor = descriptor;
+                }
+            }
+        }
+        this.customAssistant = customAssistantDescriptor;
     }
 
-    public AIAssistant getAssistant() {
-        return new AIAssistantImpl();
+    public AIAssistant getAssistant() throws DBException {
+        if (customAssistant != null) {
+            return customAssistant.createInstance();
+        } else {
+            return new AIAssistantImpl();
```

Comment:
A perfect place to throw an exception instead of silently feeding an arbitrary implementation